### ISSUE TITLE
Newest version 1.33.1 has electron_node tsserver.js eating up 100% cpu randomly

### ISSUE THREAD

**Issue by danboyle8637** [microsoft/TypeScript#31341]
So this seems to be a recurring issue over the years. But there also appears to be no definitive solution either.

Here's what my process manager is showing:
![Screen Shot 2019-04-19 at 11 14 04 AM](https://user-images.githubusercontent.com/26289514/57331112-d97a3f80-70e5-11e9-9f08-d901cb80907e.png)

Not entirely sure why. I'm not coding in Typescript... I'm not using any clear extension that's typescript related... unless ESLint does this, which I wouldnt be surprised about.

I have reverted back to version 1.32 and everything runs like a dream. 

Intellisense is FAST.
No code highlighting issues.

I just wanted to bring this back up. 

I'm working on a tight deadline for two sites so once I get these done I will try to get my tslog info... but in the meantime I wanted to see if anybody else has experienced this and found a solution that I have not come across yet. 

Thanks!

**Comment by mjbvz** [comment]
Thanks. I could repo some sluggish performance with the following steps:

1. 
    ```bsh
    git clone https://github.com/danboyle8637/thistimefitness.git
    cd thistimefitness
    npm i
    ```

1. Open folder in vscode and go to `CallToActionSection.js`


This looks like the same issue as #31302 since I usually see the poor performance in files that reference styled-components

**Comment by danboyle8637** [comment]
Just curious... why would version 1.32.3 be so fast with styled-components?

Do you recommend something else like Emotion?

**Comment by mjbvz** [comment]
See https://github.com/microsoft/TypeScript/issues/30819 for details about the root cause. You can change the TS version used in VSCode by following [these instructions](https://code.visualstudio.com/docs/typescript/typescript-compiling#_using-newer-typescript-versions)

**Comment by ahejlsberg** [comment]
Best I can tell this is fixed by #31354. I tried the setup described [above](https://github.com/microsoft/TypeScript/issues/31341#issuecomment-491386540) with #31354, and other than 10-15 seconds of CPU activity right after opening the folder, I'm not seeing much CPU load and statement completion is pretty much instantaneous.

**Comment by John-Yue** [comment]
快被这个问题折磨疯了！！！
从网上看有3中方法，
1. 设置files.exclude 和files.watcherExclude，关于内容你们肯定都看过。
2. rm -r /home/(user name)/Library/Cache/typescript/*  这个我试了一下，时灵时不灵
3. disable TypeScript/Javascript Features，这是最管用的...
